### PR TITLE
(#2860) Add new PowerShell helper function for acquiring paths from private variables

### LINKED ISSUE #2860: Add cmdlet for acquiring paths set in environment variables that should not be used directly
## Is Your Feature Request Related To A Problem? Please describe.

Some time ago we started disallowing the use of certain private environment variables when packages are pushed to Chocolatey Community Repository (such as the package folder), this leaves users without an easy way to get the information about these paths easily.

## Describe The Solution. Why is it needed?

We should add a new chocolatey helper script that can be used to get paths to folders that we allow users to use, but do not allow to be used through environment variables. Such as `ChocolateyPackageFolder` environment variable.
We should implement this as a generic helper, where the user needs to specify which type of folder path they are interested in.

## Additional Context.

N/A.

## Related Issues

N/A

### PR DESCRIPTION
**Pull request by AdmiringWorm**
## Description Of Changes

This pull request adds a new PowerShell helper that can be used going forward to get access to the private environment variable `PackagePath`, as well as to get the installation directory of Chocolatey CLI.

## Motivation and Context

There is no direct replacement for using the environment variable `$env:PackagPath` and `$env:ChocolateyPackagePath` that can be used by maintainers (at least on Chocolatey Community Repository.).
As such it makes sense to add this ability to maintainers through the use of a new cmdlet that can be used.

## Testing

1. Mainly tested by running new tests in test kitchen.

But manual testing can be done with:

1. Create a new nupkg package of the source files located in `tests/packages/test-chocolateypath`.
2. Run `choco install <PATH-TO-NUPKG>`
3. Ensure the package outputs the expected values for the package path, and the install path.
4. Run `choco uninstall test-chocolateypath`
5. Ensure the package outputs the expected values for the package path and install path both for the before modify script and the uninstall script.

## Change Types Made

* [ ] Bug fix (non-breaking change)
* [x] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)
* [ ] PowerShell code changes.

## Related Issue

Fixes #2860

## Change Checklist

* [ ] Requires a change to the documentation
* [ ] Documentation has been updated
* [x] Tests to cover my changes, have been added
* [x] All new and existing tests passed.
* [x] PowerShell v2 compatibility checked.

### CODE REVIEW

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1 [hunk 1]
```diff
@@ -0,0 +1,84 @@
+﻿# Copyright © 2017 - 2021 Chocolatey Software, Inc.
```
[the remark this answers is not in the document]
Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1:1]: I basically just copied the copyright from other files, but sure I can update this.

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1 [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            } else {
+                $result = "$env:SystemDrive\ProgramData\chocolatey"
+            }
+        }
+        Default {
+            throw "The path type $pathType is not a supported."
+        }
+    }
+
+    if ((Test-Path $result) -or ($pathType -eq 'ToolsPath')) {
```
[the remark this answers is not in the document]
Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1:81]: I originally had the support for ToolsPath as well (which basically just called the `Get-ToolsLocation`). When I removed it I forgot to remove it here as well. I'll get it removed.

[the remark this answers is not in the document]
Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1]: Fixed this up by removing the additional check for `ToolsPath`.

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyPath.ps1]: Fixed up the copyright here.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@AdmiringWorm thanks for getting this updated!

**Pull request merged**